bob-apps/bin/bobd.rs

```diff
@@ -34,7 +34,26 @@ async fn main() {
     log4rs::init_file(node.log_config(), Default::default()).expect("can't find log config");
 
     let mut mapper = VirtualMapper::new(&node, &cluster).await;
-    let mut addr = node.bind().to_socket_addrs().unwrap().next().unwrap();
+
+    let mut addr = match (
+        node.bind_to_ip_address(),
+        node.bind().parse(),
+        port_from_address(node.bind().as_str()),
+    ) {
+        (Some(addr1), Ok(addr2), _) => {
+            if addr1 == addr2 {
+                Some(addr1)
+            } else {
+                log::error!("Addresses provided in node config and cluster config are not equal: {:?} != {:?}", addr1, addr2);
+                None
+            }
+        }
+        (Some(addr), _, _) => Some(addr),
+        (_, Ok(addr), _) => Some(addr),
+        (_, _, Some(port)) => Some(SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), port)),
```

Comment:
`SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), port)` occured multiple times in the code, prob it's better to add some function for it (since it's quite a long line)

Reply:
Or `type`

Reply:
Function created